src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -506,6 +509,10 @@ extern "C" void PalAttachThread(void* thread)
 extern "C" bool PalDetachThread(void* thread)
 {
     UNREFERENCED_PARAMETER(thread);
+    if (g_threadDetachCallback != nullptr)
```

Comment:
In this case it would be hard to guarantee that the callback happens before PalDetachThread.  Do you have good ideas how to be notified on thread termination?  We used pthread_key_create in the past, however, it used to be called too late, e.g., tls_CurrentThread was already destroyed (according to https://github.com/dotnet/corert/commit/6eedad45bcb621c6d26aafa4ec47d37cdbc1204e).  Though I guess we might reinstall the destroyed value back into TLS like CoreCLR does...